test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeConfiguration.cs

```diff
@@ -23,9 +23,127 @@ public class EdgeConfiguration
+                        {
+                            envObj.Add(pair.Item1, JToken.FromObject(new { value = pair.Item2 }));
+                        }
+                    });
+
+                return this;
+            }
+
+            public Module WithProxy(Option<Uri> proxy, Protocol protocol)
+            {
+                proxy.ForEach(
+                    p =>
+                    {
+                        string proxyProtocol;
+                        switch (protocol)
+                        {
+                            case Protocol.Amqp:
+                                proxyProtocol = Protocol.AmqpWs.ToString();
+                                break;
+                            case Protocol.Mqtt:
+                                proxyProtocol = Protocol.MqttWs.ToString();
+                                break;
+                            case Protocol.AmqpWs:
+                            case Protocol.MqttWs:
+                                proxyProtocol = protocol.ToString();
+                                break;
+                            default:
+                                throw new ArgumentException("Unknown protocol");
+                        }
+
+                        // If UpstreamProtocol was already set in this config to a compatible value,
+                        // remove it so we can add the proxy-compatible value. If it's not compatible,
+                        // throw an error. The caller will need to fix the conflict in their code.
+                        Option<string> upstreamProtocol = this.GetEnvironmentVariable("UpstreamProtocol");
+                        if (upstreamProtocol.Exists(u => u == protocol.ToString() || u == proxyProtocol))
+                        {
+                            this.RemoveEnvironmentVariable("UpstreamProtocol");
+                        }
+                        else
+                        {
+                            upstreamProtocol.ForEach(
+                                u =>
+                                {
+                                    string message = $"Setting \"UpstreamProtocol\" to \"{proxyProtocol}\"" +
+                                                     $"would overwrite incompatible value \"{u}\"";
+                                    throw new ArgumentException(message);
+                                });
+                        }
+
+                        this.WithEnvironment(
+                            new[]
+                            {
+                                ("https_proxy", p.ToString()),
+                                ("UpstreamProtocol", proxyProtocol)
+                            });
+                    });
+
+                return this;
+            }
+
+            Option<string> GetEnvironmentVariable(string key)
+            {
+                Option<string> value = Option.None<string>();
+
+                this.config.UpdateModule(
```

Comment:
This name is named as "GetEnvironmentVariable" but it calls config.UpdateModule; what does config.UpdateModule does?

Reply:
UpdateModule became overloaded. My most recent commits to introduce a builder API deleted this class.